src/sdl2/render.rs

```diff
@@ -492,30 +496,81 @@ impl TextureCreator {
     }
 }
 
+pub enum TargetKind {
+    Canvas,
+    Texture,
+}
+
+pub trait Target {
+    fn kind(&self) -> TargetKind;
+}
+
+pub struct CanvasTarget;
+
+impl Target for CanvasTarget {
+    fn kind(&self) -> TargetKind {
+        TargetKind::Canvas
+    }
+}
+
+pub struct TextureTarget<'r, 't> {
+    _renderer_marker: PhantomData<&'r ()>,
+    _texture_marker: PhantomData<&'t ()>,
+}
+
+impl<'r, 't> Target for TextureTarget<'r, 't> {
+    fn kind(&self) -> TargetKind {
+        TargetKind::Texture
+    }
+}
+
+pub type CanvasRenderer = Renderer<CanvasTarget>;
+
 /// Render target methods
-impl Renderer {
+impl CanvasRenderer {
     /// Determine whether a window supports the use of render targets.
     pub fn render_target_supported(&self) -> bool {
         unsafe { ll::SDL_RenderTargetSupported(self.context.raw) == 1 }
     }
 
-    /// Gets the render target handle.
-    ///
-    /// Returns `None` if the window does not support the use of render targets.
-    pub fn render_target(&self) -> Option<RenderTarget> {
+    /// Sets the render target to the provided texture
+    /// Returns a handle for rendering methods to the target
+    /// Returns `SdlError` if the window does not support the use of render targets,
+    /// or if SDL2 returned with an error code
+    /// The texture must be created with the texture access: `sdl2::render::TextureAccess::Target`.
+    pub fn target<'r, 't, 'a>(&'r mut self,
+                              texture: &'t mut Texture<'a>)
+                              -> Result<TextureRenderer<'r, 't>, IntegerOrSdlError> {
         if self.render_target_supported() {
-            Some(RenderTarget {
-                     raw: self.context.raw,
-                     _marker: PhantomData,
-                 })
+            unsafe { self.raw_target(texture.raw) }?;
+            Ok(Renderer {
+                   context: self.context.clone(),
+                   target: TextureTarget {
+                       _renderer_marker: PhantomData,
+                       _texture_marker: PhantomData,
+                   },
+               })
         } else {
-            None
+            Err(IntegerOrSdlError::SdlError("window does not support the use of render targets"
+                                                .into()))
+        }
+    }
+}
+
+impl<T: Target> Renderer<T> {
+    unsafe fn raw_target(&mut self,
```

Comment:
`set_raw_target` would be better here as well. Same comment for the Error part, we will never have an Integer error here, so one variant will effectively be useless.